controllers/usernamespace/controller.go

```diff
@@ -0,0 +1,393 @@
+			if key.Name == "" || (key.Name == k.Name && key.Namespace == k.Namespace) {
+				return &v
+			}
+			return nil
+		}
+	}
+
+	ret, ok := instances[key]
+
+	if ok {
+		return &ret
+	} else {
+		return nil
+	}
+}
+
+func (r *CheUserNamespaceReconciler) reconcileSelfSignedCert(ctx context.Context, deployContext *deploy.DeployContext, targetNs string, checluster *v2alpha1.CheCluster) error {
+	targetCertName := prefixedName(checluster, "server-cert")
+
+	delSecret := func() error {
+		_, err := deploy.Delete(deployContext, client.ObjectKey{Name: targetCertName, Namespace: targetNs}, &corev1.Secret{})
+		return err
+	}
+
+	cheCert := &corev1.Secret{}
+	if err := r.client.Get(ctx, client.ObjectKey{Name: deploy.CheTLSSelfSignedCertificateSecretName, Namespace: checluster.Namespace}, cheCert); err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+		// There is not self-signed cert in the namespace of the checluster, so we have nothing to copy around
+		return delSecret()
+	}
+
+	if _, ok := cheCert.Data["ca.crt"]; !ok {
+		// the secret doesn't contain the certificate. bail out.
+		return delSecret()
+	}
+
+	targetCert := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      targetCertName,
+			Namespace: targetNs,
+			Labels: defaults.AddStandardLabelsForComponent(checluster, userSettingsComponentLabelValue, map[string]string{
+				constants.DevWorkspaceMountLabel: "true",
+			}),
+			Annotations: map[string]string{
+				constants.DevWorkspaceMountAsAnnotation:   "file",
+				constants.DevWorkspaceMountPathAnnotation: "/tmp/che/secret/",
+			},
+		},
+		Data: map[string][]byte{
+			"ca.crt": cheCert.Data["ca.crt"],
+		},
+	}
+
+	_, err := deploy.DoSync(deployContext, targetCert, deploy.SecretDiffOpts)
+	return err
+}
+
+func (r *CheUserNamespaceReconciler) reconcileTrustedCerts(ctx context.Context, deployContext *deploy.DeployContext, targetNs string, checluster *v2alpha1.CheCluster) error {
+	targetConfigMapName := prefixedName(checluster, "trusted-ca-certs")
+
+	delConfigMap := func() error {
+		_, err := deploy.Delete(deployContext, client.ObjectKey{Name: targetConfigMapName, Namespace: targetNs}, &corev1.Secret{})
+		return err
+	}
+
+	sourceMap := &corev1.ConfigMap{}
+	if err := r.client.Get(ctx, client.ObjectKey{Name: deploy.CheAllCACertsConfigMapName, Namespace: checluster.Namespace}, sourceMap); err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+
+		return delConfigMap()
+	}
+
+	targetMap := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      targetConfigMapName,
+			Namespace: targetNs,
+			Labels: defaults.AddStandardLabelsForComponent(checluster, userSettingsComponentLabelValue, map[string]string{
+				constants.DevWorkspaceMountLabel:              "true",
+				"config.openshift.io/inject-trusted-cabundle": "true",
```

Comment:
You don't have to add this label. `deploy.CheAllCACertsConfigMapName` already contains all OpenShift certificates.